### DISCUSSION TITLE
Export of trained model to pytorch

### DISCUSSION THREAD

**Discussion by evenlund** [lululxvi/deepxde#1697] [Q&A]
I have read the answer to "How can we save model so for multiple predictions #54", and see that it is possible to save and open a trained model. Is it possible somehow to export a model trained in deepxde with pytorch backend and open it in pytorch?

**Comment by xuliang5115** [comment]
I think you should attach more information about this sentence to avoid misunderstanding.
_"Is it possible somehow to export a model trained in deepxde with pytorch backend and open it in pytorch?"_
1. Restore `trained model` only with `pytorch`, without `deepxde`,
2. Or there is a typo, you means to restore `trained model` with another backend like `tensorflow`?

**Reply by evenlund** [reply]
I meant what you suggest in 1. "Restore trained model only with pytorch, without deepxde". The reason I wrote the question like this was that I thought you needed to have pytorch set as backend first to later make it possible to import/restore in pytorch.

Anyways I think I figured it out myself, sort of: 

So if you train your model while saving with deepxde like this:
`losshistory, train_state = model.train(iterations = 15000, model_save_path='./heat_deepxde')`

You are able to restore the trained pytorch model like this:
`model_torch=dde.nn.pytorch.fnn.FNN([2] + [20]*3+ [1], "tanh", "Glorot normal") #Need to specify nn-architecture first. OK if you set up the model`
`checkpoint = torch.load('./heat_deepxde-15000.pt') #loading deepxde checkpoint dict containing trained model and optimizer with settings`
`model_torch.load_state_dict(checkpoint['model_state_dict']) #restoring model with saved state`
`optimizer_state_dict=checkpoint['optimizer_state_dict'] #Unable to restore the optimizer so far`
`model_torch.eval() #Unneccesary?`
